src/m3/deserializer.ts

```diff
@@ -0,0 +1,200 @@
+import {
+    Concept,
+    ConceptInterface,
+    Containment,
+    Datatype,
+    Feature,
+    FeaturesContainer,
+    Link,
+    M3Concept,
+    Metamodel,
+    MetamodelElement,
+    PrimitiveType,
+    Property,
+    Reference
+} from "./types.ts"
+import {SingleRef} from "../references.ts"
+import {Id, Node} from "../types.ts"
+import {SerializedNode} from "../serialization.ts"
+import {metaConcepts, metaFeatures} from "./self-definition.ts"
+
+
+export const deserializeMetamodel = (serializedNodes: SerializedNode[]): Metamodel => {
+
+    const metamodelSerNode = serializedNodes.find(({type}) => type === metaConcepts.metamodel.id)
```

Comment:
should we check there is exactly one?

Reply:
In general: no. A serialized model could contain multiple roots, AFAIA. If we want to constrain this for (serializations of) metamodels we should make that explicit.